pkg/operator/controllers/node/node_controller.go

```diff
@@ -0,0 +1,155 @@
+	}
+
+	if !isDraining(node) {
+		// we're not draining: ensure our annotation is not set and return
+		if getAnnotation(&node.ObjectMeta, annotationDrainStartTime) == "" {
+			return reconcile.Result{}, nil
+		}
+
+		delete(node.Annotations, annotationDrainStartTime)
+
+		_, err = r.kubernetescli.CoreV1().Nodes().Update(ctx, node, metav1.UpdateOptions{})
+		if err != nil {
+			r.log.Error(err)
+		}
+
+		return reconcile.Result{}, err
+	}
+
+	// we're draining: ensure our annotation is set
+	t, err := time.Parse(time.RFC3339, getAnnotation(&node.ObjectMeta, annotationDrainStartTime))
+	if err != nil {
+		t = time.Now().UTC()
+		setAnnotation(&node.ObjectMeta, annotationDrainStartTime, t.Format(time.RFC3339))
+
+		node, err = r.kubernetescli.CoreV1().Nodes().Update(ctx, node, metav1.UpdateOptions{})
+		if err != nil {
+			r.log.Error(err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// if our deadline hasn't expired, requeue ourselves and return
+	deadline := t.Add(gracePeriod)
+	now := time.Now()
+	if deadline.After(now) {
+		return reconcile.Result{
+			RequeueAfter: deadline.Sub(now),
+		}, err
+	}
+
+	// drain the node disabling eviction
+	err = drain.RunNodeDrain(&drain.Helper{
+		Client:              r.kubernetescli,
+		Force:               true,
+		GracePeriodSeconds:  -1,
+		IgnoreAllDaemonSets: true,
+		Timeout:             60 * time.Second,
+		DeleteLocalData:     true,
+		DisableEviction:     true,
+		OnPodDeletedOrEvicted: func(pod *corev1.Pod, usingEviction bool) {
+			r.log.Print("deleted pod %s/%s", pod.Namespace, pod.Name)
+		},
+		Out:    r.log.Writer(),
+		ErrOut: r.log.Writer(),
+	}, request.Name)
+	if err != nil {
+		r.log.Error(err)
+		return reconcile.Result{}, err
+	}
+
+	// ensure our annotation is not set and return
+	delete(node.Annotations, annotationDrainStartTime)
+
+	_, err = r.kubernetescli.CoreV1().Nodes().Update(ctx, node, metav1.UpdateOptions{})
+	if err != nil {
+		r.log.Error(err)
+	}
+
+	return reconcile.Result{}, err
+}
+
+// SetupWithManager setup our mananger
+func (r *NodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Node{}).
+		Named(controllers.NodeControllerName).
+		Complete(r)
+}
+
+func getAnnotation(m *metav1.ObjectMeta, k string) string {
+	if m.Annotations == nil {
+		return ""
+	}
+
+	return m.Annotations[k]
+}
+
+func setAnnotation(m *metav1.ObjectMeta, k, v string) {
+	if m.Annotations == nil {
+		m.Annotations = map[string]string{}
+	}
+
+	m.Annotations[k] = v
+}
+
+func isDraining(node *corev1.Node) bool {
+	return ready.NodeIsReady(node) &&
+		node.Spec.Unschedulable &&
+		getAnnotation(&node.ObjectMeta, annotationCurrentConfig) != "" &&
+		getAnnotation(&node.ObjectMeta, annotationDesiredConfig) != "" &&
+		getAnnotation(&node.ObjectMeta, annotationCurrentConfig) != getAnnotation(&node.ObjectMeta, annotationDesiredConfig) &&
+		getAnnotation(&node.ObjectMeta, annotationState) == stateWorking
```

Comment:
There may be a possibility where the MCO is not working because it failed to drain.  That could result in the 'force drain' by this operator not running.  May need to exclude the `stateWorking` check in this function

Reply:
I think I saw this today on a 4.4 cluster - I had to manually kill the machine config daemon.  Is that what you mean?  I'm not sure in what cases this happens and it's a big hammer to apply this across the board.  Do you have a clearer view of the circumstances when this happens?

Reply:
Yes this is the case.  When the config for a machine updates, the MCO will attempt to drain a node.  If this fails, it will keep trying with an exponential backoff policy.  During failure, the annotation on the node being drained with exponential backoff is set to `machineconfiguration.openshift.io/state: Degraded`.  Even though this state is set to `Degraded`, it will continue to attempt to drain the node. 

Because of this, the `isDraining` function should be updated as follows as there is a possibility during the grace period that the MCO times out draining and the annotation is set to `Degraded` and not `Working`.  
```go
func isDraining(node *corev1.Node) bool {
	return ready.NodeIsReady(node) &&
		node.Spec.Unschedulable &&
		getAnnotation(&node.ObjectMeta, annotationCurrentConfig) != "" &&
		getAnnotation(&node.ObjectMeta, annotationDesiredConfig) != "" &&
		getAnnotation(&node.ObjectMeta, annotationCurrentConfig) != getAnnotation(&node.ObjectMeta, annotationDesiredConfig) &&
		(getAnnotation(&node.ObjectMeta, annotationState) == stateWorking || 
		getAnnotation(&node.ObjectMeta, annotationState) == "Degraded" )
```

Reply:
The problem is that yesterday I saw Degraded errors for other reasons (https://access.redhat.com/solutions/4951051) and killing the pod wouldn't have helped there.  Perhaps we could test something like `|| (degraded && kubelet version corresponds to 4.4)` but I think we need more data